includes/class-sensei-course.php

```diff
@@ -3446,34 +3446,45 @@ private function is_front_page( $query ) {
 	 * @since 1.9.10
 	 */
 	public static function prerequisite_complete_message() {
```

Comment:
I wonder if we could replace the event where it's called (`sensei_single_course_content_inside_before`) with a generic event and use the same method for both legacy and courses with blocks. So we don't duplicate this logic, and it becomes generic for the block too instead of being handled in the render method. WDYT?

Reply:
Isn't this a bit dangerous to break old sites?

Reply:
Hum, IMHO if we test in the main themes and it works, I think it worths it, so we get rid of some legacy maintenance. :)

Reply:
Looking the 577-gh-Automattic/sensei-wc-paid-courses, another advantage would be that we don't need to handle it again in the purchase block.

Reply:
The thing is that if we move this out of `render_callback`, we would need to call `Sensei_Course::is_prerequisite_complete` in both places as we render the button disabled in that case. So in a way I think that the current solution is better as all 'prerequisite not complete' logic is in one place. 

Also removing the action from `sensei_single_course_content_inside_before` does not make the code less complicated so we should probably avoid the risk of braking old sites.